crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -104,7 +106,9 @@ fn check_trait_method_implementation_return_type(
     impl_method: &NoirFunction,
     trait_name: &str,
 ) -> Result<(), DefCollectorErrorKind> {
-    if expected_return_type.is_equivelent(&impl_method.def.return_type) {
+    if UnresolvedType::from(expected_return_type.clone())
+        == UnresolvedType::from(impl_method.def.return_type.clone())
```

Comment:
Using `UnresolvedType::from` requires cloning the type. To avoid this, we can add a method to get a reference to an UnresolvedType instead.

Reply:
I applied the suggested changes. Btw, I was wondering why the `from_int_type` is not an implementation of from trait. Now, I know. Thanks!